src/util/collections.common.tests.ts

```diff
@@ -0,0 +1,150 @@
+import { MockedResponse } from '@apollo/react-testing';
+import { screen, waitFor } from '@testing-library/react';
+import range from 'lodash/range';
+
+import {
+  CollectionListDocument,
+  CollectionListQueryVariables,
+  CollectionListResponse,
+} from '../generated/graphql';
+import { fakeCollections } from './mockDataUtils';
+import { render } from './testUtils';
+
+export const getCollectionQueryListMocks = (
+  collections: CollectionListResponse,
+  variables?: CollectionListQueryVariables
+): MockedResponse[] => [
+  {
+    request: {
+      query: CollectionListDocument,
+      variables,
+    },
+    result: { data: { collectionList: collections } },
+  },
+];
+
+const renderAndWaitForComponentToBeLoaded = async (
+  component: React.ReactElement,
+  collections: CollectionListResponse,
+  mocks: MockedResponse[] = [],
+  variables?: CollectionListQueryVariables
+) => {
+  const collectionMocks = getCollectionQueryListMocks(collections, variables);
+  render(component, { mocks: [...collectionMocks, ...mocks] });
+
+  await waitFor(() => {
+    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+  });
+};
+
+const renderComponentAndExpectOddCollectionsToBePresent = async (
+  component: React.ReactElement,
+  collections: CollectionListResponse,
+  collectionNames: string[],
+  mocks: MockedResponse[] = [],
+  variables?: CollectionListQueryVariables
+) => {
```

Comment:
I would prefer these functions would receive one object as parameter instead of 4-5 parameters, It would easier in the future to pass parameters because the ordering doesn't have to be anything specific.

With destructuring passing object is handy

Reply:
Yes it makes it also more readable at least when browsing github. Done.